cdap-ui/app/cdap/components/AbstractWidget/AbstractMultiRowWidget/AbstractRow.tsx

```diff
@@ -87,26 +88,27 @@ export default class AbstractRow<
   };
 
   public render() {
-    const { errors, value } = this.props;
+    const { errors, value, dataCy, index } = this.props;
     let errorMsg = null;
     if (errors && value) {
       const errorObj = errors.find((error: IErrorObj) => error.element === value);
       if (errorObj) {
         errorMsg = errorObj.msg;
       }
     }
+
     return (
       <React.Fragment>
-        <div className={this.props.classes.root}>
+        <div className={this.props.classes.root} data-cy={`${index}`}>
           {this.renderInput()}
 
           <If condition={!this.props.disabled}>
             <React.Fragment>
-              <IconButton onClick={this.props.addRow}>
+              <IconButton onClick={this.props.addRow} className="add-row">
```

Comment:
Any reason why you're using `className` instead of `data-cy` here and in other places in this PR?